### PR TITLE
Ports Bay12 Posibrains[Ask me before Merge.]

### PR DESCRIPTION
**Pull request by RemieRichards**
Posibrains are MMIs that can be made, and when clicked in hand request a ghost to inhabit them. They work EVERYWHERE an MMI works, Borgs, AI, Mechs

### CODE REVIEW

**Comment by RemieRichards** [comment]
Woopsie Commits were done on the Github editor, hence the tiny changes per commit. :)

**Comment by RemieRichards** [comment]
Just dealing with some HUD based runtimes atm.

**Comment by RemieRichards** [comment]
Runtimes removed.

**Comment by ghost** [comment]
This would be good for mommi/keeper/drone construction rather than a fabricator shitting them out.

**Comment by RemieRichards** [comment]
That's true @paprka

**Comment by GunHog** [comment]
Build Positronic brain. Build a drone frame from the protolathe / fabricator. Activate the brain and get a ghost. Shove brain into drone frame. You now have a little drone pet! Of course, you could make it into a borg, AI, or mech. It is nice to have options.

**Comment by RemieRichards** [comment]
@GunHog that'll come with the little improvements to Drones and we go along.

**Comment by Dorsisdwarf** [comment]
Does it make it very, very explicit in the text when you enter a posibrain that you know NOTHING your old character knew, absolutely, clean start, no you may not go secbot and just "happen" to randomly search the guy you knew kill you?

**Comment by RemieRichards** [comment]
@Dorsisdwarf Good catch, I forgot about Wiping past lives with wipe_memory() etc.

**Comment by ghost** [comment]
Poll it. We can discuss opinions on porting posibrains in a forum thread, not here. Or IRC.

**Comment by GunHog** [comment]
Did you conflict yourself, RR?

**Comment by RemieRichards** [comment]
yeah, it's why the PR has such an odd [Tag] on it, I knew it would do this so I prioritised the feature which originally had less bugs, in this case, Drones.

File: code/modules/mob/living/carbon/brain/posibrain.dm [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+/obj/item/device/mmi/posibrain/examine()
+
+	set src in oview()
+
+	if(!usr || !src)	return
+	if( (usr.sdisabilities & BLIND || usr.blinded || usr.stat) && !istype(usr,/mob/dead/observer) )
+		usr << "<span class='notice'>Something is there but you can't see it.</span>"
+		return
+
+	var/msg = "<span class='info'>*---------*\nThis is \icon[src] \a <EM>[src]</EM>!\n[desc]\n"
```
Comment (JJRcop) [inline code/modules/mob/living/carbon/brain/posibrain.dm:113]: Isn't `\a` redundant?

Comment (RemieRichards) [inline code/modules/mob/living/carbon/brain/posibrain.dm:113]: Posibrain names aren't all supportive of A or all supportive of AN so \a is fine there.

**Comment by GunHog** [comment]
Plz finish :+1:

**Comment by RemieRichards** [comment]
Note, MrPerson is fixing some stuff with mechs and He says he'll try and help me with the Mech-MMI runtimes me and @GunHog found.

**Comment by JJRcop** [comment]
@RemieRichards poke :point_right:

**Comment by GunHog** [comment]
Yes, fix conflicts and merge plz (After the runtime fix)

**Comment by GunHog** [comment]
If this method is not suitable, you may consider a special MMI (Machine-Machine Interface) that would accept pAIs. That would have little difference behavior-wise with cyborgs and AIs, but the concern would be using them as mech pilots. (Such as a Durand army slaved to a traitor with orders to murderbone) Positronic mech pilots could never be an antagonist, while a pAI could be one by proxy.

**Comment by RemieRichards** [comment]
GunHog. a Special MMI is EXACTLY what a Posibrian is, It would run into exactly the same bugs, Probably MORE with it being a pAI as well.

File: code/modules/mob/living/carbon/brain/posibrain.dm (hunk 1 above)
Comment (JJRcop) [inline code/modules/mob/living/carbon/brain/posibrain.dm:113]: No no, I mean, just not having A, AN, or `\a`, because isn't `[src]` supposed to insert it automatically?

Comment (RemieRichards) [inline code/modules/mob/living/carbon/brain/posibrain.dm:113]: Uh, the only uses of A and AN are to correctly insert A, AN etc in the sentence when you don't know what it is, as in [src] or [exampleobj], I don't believe [src] or [exampleobj] or whatever automatically place A/AN because that reduces the uses of A and AN down to "this is \a bike" which will always be "this is a bike".

tl;dr I think you're wrong, and if you're not wrong then \a and \an are stupid.

Comment (JJRcop) [inline code/modules/mob/living/carbon/brain/posibrain.dm:113]: Yeah, I am wrong, sorry. I tested by removing `\a` from `/atom/verb/examine()`, and it said "That's the fire alarm.". So `[src]` automatically inserts "the" or "The", not "a" or "an".

**Comment by GunHog** [comment]
RR Plz Finnish K Thx L0L

**Comment by GunHog** [comment]
Agreed, this has the added bonus of making pAIs more desired by ghosted players. It is not a bad way to get AIs playing as well. Artyom does often suffer the lack of AI players during lowpop.
